org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java

```diff
@@ -11644,6 +11647,15 @@ public void unlikelyArgumentType(Expression argument, MethodBinding method, Type
 			argument.sourceEnd);
 }
 
+public void unlikelyEqualExpressionArgumentType(EqualExpression comparison, char[] expressionType) {
+	this.handle(
+			IProblem.UnlikelyEqualExpressionArgumenType,
+			NoArgument,
```

Comment:
If we agree on mentioning the expression type, then also `problemArguments` must be used (one is the short form of the other).

Reply:
What is the purpose of the `problemArguments` parameter (as in how is it used)? Should it have the same form as the `messageArguments`, but instead use readableName for shortReadableName?